src/main/java/com/ericsson/ei/subscriptionhandler/SubscriptionValidator.java

```diff
@@ -0,0 +1,137 @@
+//		List<Requirement> reqList = subscription.getRequirements();
+//		for (int i=0; i < reqList.size(); i++) {
+//			this.validateJmespath(reqList.get(i).getConditions().get(0).getJmespath());
+//		}
+		LOG.info("Validating of subscription " + subscription.getSubscriptionName() + " finished successfully.");
+	}
+	
+	/*
+	 * Validation of subscriptionName parameter
+	 * Throws SubscriptionValidationException if validation of the parameter fails due to wrong
+	 * format of parameter.
+	 * 
+	 * @param subscriptionName
+	 */
+	public void validateSubscriptionName(String subscriptionName) throws SubscriptionValidationException {
+		
+		String regex = "^[A-Za-z0-9_]+$";
+		
+		if (!Pattern.matches(regex, subscriptionName)) {
+			throw new SubscriptionValidationException("Wrong format of SubscriptionName: " + subscriptionName);
+		}	
+	}
+	
+	/*
+	 * Validation of notificationMessage parameter
+	 * Throws SubscriptionValidationException if validation of the parameter fails due to wrong
+	 * format of parameter.
+	 * 
+	 * @param notificationMessage
+	 */
+	public void validateNotificationMessage(String notificationMessage) throws SubscriptionValidationException {
+		
+		String regex = "^[A-Za-z0-9@.]+$";
+		
+		if (!Pattern.matches(regex, notificationMessage)) {
+			throw new SubscriptionValidationException("Wrong format of NotificationMessage: " + notificationMessage);
+		}	
+	}
+	
+	/*
+	 * Validation of notificationMeta parameter
+	 * Throws SubscriptionValidationException if validation of the parameter fails due to wrong
+	 * format of parameter.
+	 * 
+	 * @param notificationMeta
+	 */
+	public void validateNotificationMeta(String notificationMeta) throws SubscriptionValidationException {
+		String regex = ".*[\\s].*";
+		
+		if (Pattern.matches(regex, notificationMeta)) {
+			throw new SubscriptionValidationException("Wrong format of NotificationMeta: " + notificationMeta);
+		}
+	}
+	
+	/*
+	 * Validation of notificationType parameter
+	 * Throws SubscriptionValidationException if validation of the parameter fails due to wrong
+	 * format of parameter.
+	 * 
+	 * @param notificationType
+	 */
+	public void validateNotificationType(String notificationType) throws SubscriptionValidationException {
+		String regexMail = "[\\s]*MAIL[\\\\s]*";
+		String regexRestPost = "[\\s]*REST_POST[\\\\s]*";
+		
+		if (!(Pattern.matches(regexMail, notificationType) || Pattern.matches(regexRestPost, notificationType))) {
+			throw new SubscriptionValidationException("Wrong format of NotificationType: " + notificationType);
+		}
+	}
+	
+	/*
+	 * Validation of repeat parameter
+	 * Throws SubscriptionValidationException if validation of the parameter fails due to wrong
+	 * format of parameter.
+	 * 
+	 * @param repeat
+	 */
+	public void validateRepeat(Boolean repeat) throws SubscriptionValidationException {
+		if (!(repeat instanceof Boolean)) {
+			throw new SubscriptionValidationException("Wrong format of Repeat: " + repeat.toString());
+		}
+		
+	}
```

Comment:
Shall I remove this validator. It will never fail.
I guess it should fail earlier when EI Backend receives the EI RestApi request and decode it from JSON to Subscription Class, to the boolean repeat parameter/value. In this code it will always be boolean and never fail.
I feels like this check is unnecessary and can be removed.